x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java

```diff
@@ -55,37 +55,72 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import static org.elasticsearch.common.settings.Setting.Property.Dynamic;
+import static org.elasticsearch.common.settings.Setting.Property.NodeScope;
+
 public class APMTracer extends AbstractLifecycleComponent implements TracingPlugin.Tracer {
 
     public static final CapturingSpanExporter CAPTURING_SPAN_EXPORTER = new CapturingSpanExporter();
 
+    static final Setting<Boolean> APM_ENABLED_SETTING = Setting.boolSetting("xpack.apm.enabled", false, Dynamic, NodeScope);
```

Comment:
I know the other two settings are also under `xpack.apm` but this name worries me a bit - we're not disabling APM, just tracing ES internals to APM. We should move to a more specific namespace, maybe `xpack.tracing.apm_exporter.*`. We can do this in a follow-up, no need to combine these changes here.